public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts

```diff
@@ -14,8 +22,69 @@ export interface DashboardScenePageState {
   loadError?: string;
 }
 
+export const DASHBOARD_CACHE_TTL = 2000;
+
+interface DashboardCacheEntry {
+  dashboard: DashboardDTO;
+  ts: number;
+}
 export class DashboardScenePageStateManager extends StateManagerBase<DashboardScenePageState> {
   private cache: Record<string, DashboardScene> = {};
+  // This is a simplistic, short-term cache for DashboardDTOs to avoid fetching the same dashboard multiple times across a short time span.
+  private dashboardCache: Map<string, DashboardCacheEntry> = new Map();
+
+  // To eventualy replace the fetchDashboard function from Dashboard redux state management.
+  // For now it's a simplistic version to support Home and Normal dashboard routes.
+  public async fetchDashboard(uid: string) {
+    const cachedDashboard = this.getFromCache(uid);
+
+    if (cachedDashboard) {
+      return cachedDashboard;
+    }
+
+    let rsp: DashboardDTO | undefined;
+
+    try {
+      if (uid === DashboardRoutes.Home) {
```

Comment:
nipit: this will break if someone created dashboard with  `home-dashboard` as the uid? think creating a separate fetchHome function could be better

Reply:
Yeah, that's a possible edge case! Will address in a followup.